src/xdist/remote.py

```diff
@@ -72,6 +73,12 @@ def __init__(self, config, channel):
     def _make_queue(self):
         return self.channel.gateway.execmodel.queue.Queue()
 
+    def _get_next_item_index(self):
```

Comment:
It would be nice to add a docstring explaining why the retry is needed.

Reply:
Will `RETRY_MARK` -> `QUEUE_CHANGED_MARK` make the code more self-explanatory?

Reply:
Ahh I think so. I would still add a docstring describing the issue however.